plugins/plugin-chart-echarts/src/Gauge/transformProps.ts

```diff
@@ -0,0 +1,197 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  QueryFormMetric,
+  ChartProps,
+  CategoricalColorNamespace,
+  CategoricalColorScale,
+  DataRecord,
+  getNumberFormatter,
+  getMetricLabel,
+} from '@superset-ui/core';
+import { EChartsOption, GaugeSeriesOption } from 'echarts';
+import { GaugeDataItemOption } from 'echarts/types/src/chart/gauge/GaugeSeries';
+import { parseNumbersList } from '../utils/controls';
+import { DEFAULT_FORM_DATA as DEFAULT_GAUGE_FORM_DATA, EchartsGaugeFormData } from './types';
+
+// https://stackoverflow.com/a/10284006/2858258
+const zip = (rows: any[][]) => rows[0].map((_, c) => rows.map(row => row[c]));
+
+const setIntervalBoundsAndColors = (
+  intervals: string,
+  intervalColorIndices: string,
+  colorFn: CategoricalColorScale,
+  normalizer: number,
+): Array<[number, string]> => {
+  const intervalBoundsNonNormalized = parseNumbersList(intervals, ',') || [];
+  const intervalColorIndicesArray = parseNumbersList(intervalColorIndices, ',') || [];
+  const intervalBounds = intervalBoundsNonNormalized.map(bound => bound / normalizer);
+  const intervalColors = intervalColorIndicesArray.map(ind => colorFn.colors[ind - 1]);
+  const intervalBoundsAndColors = zip([intervalBounds, intervalColors]) as Array<[number, string]>;
+  return intervalBoundsAndColors;
```

Comment:
I think this works equally well:
```suggestion
  return intervalBounds.map((val, idx) => [val, intervalColors[idx]]);
```
Btw, I noticed that when typing the interval colors/bounds, if the parse isn't successful, the chart errors out and you have to refresh your browser to get it to work again. It might be a good idea to make this more resilient to errors (adding a try-catch to avoid having `parseNumbersList` throwing an error), and also perhaps default to sequentially stepping through the available colors if none are specified. So something like this may be preferable:

```javascript
  return intervalBounds.map((val, idx) => {
    const color = intervalColors[idx];
    return [val, color ?? colorFn.colors[idx]];
  });
};
```